source/game.database.bmx

```diff
@@ -2359,6 +2365,57 @@ Type TDatabaseLoader
 		If Not foundEntry Then Return Null
 		Return localized
 	End Function
+
+	Function onSetLanguage:Int(triggerEvent:TEventBase)
+		Local lang:String = triggerEvent.GetData().GetString("languageCode", "en")
+
+		'first restore default
+		If lang <> "en" Then SetLanguage("en")
+		SetLanguage(lang)
+	End Function
+
+	Function SetLanguage(lang:String)
+		Local baseDir:String = "res/database/Default/"
+		Local file:String = baseDir+"lang/"+lang+".xml"
+
+		Local xml:TXmlHelper = TXmlHelper.Create(file)
+		Local nodeAllPersons:TxmlNode
+		nodeAllPersons = xml.FindRootChild("persons")
+		For Local nodePerson:TxmlNode = EachIn xml.GetNodeChildElements(nodeAllPersons)
+			If nodePerson.getName() <> "person" Then Continue
+			Local data:TData = New TData
+			xml.LoadValuesToData(nodePerson, data, ["id","first_name", "last_name", "nick_name"])
+			Local id:String=data.GetString("id")
```

Comment:
eventuell sollten wir in der DB die Dinge auf "GUID" umschreiben - es wirkt komisch, wenn wir hier "id" einladen, die Variable "id" nennen - und dann `getByGUID(id)` aufrufen.

Ist aber nix, was dieser PR angehen sollte.

Reply:
Für neue Formate sollte man das dann gleich machen. Ich passe das hier noch an.